### DISCUSSION TITLE
Exception Value cannot be null or empty. (Parameter 'areaName')

### DISCUSSION THREAD

**Discussion by omarhassani** [OrchardCMS/OrchardCore#10033] [Q&A]
### To Reproduce
1. Use OrchardCore.Samples - https://github.com/OrchardCMS/OrchardCore.Samples
2. Replace ConfigureServices with this code in Startup.cs

```C#
            services.AddMvc(options => options.EnableEndpointRouting = false);
            services.AddAuthentication(AzureADDefaults.AuthenticationScheme)
            .AddAzureAD(options => Configuration.Bind("AzureAd", options));
            services.AddOrchardCore().AddMvc().ConfigureServices(t =>
            {
                t.AddAuthentication(AzureADDefaults.AuthenticationScheme)
                    .AddAzureAD(options => Configuration.Bind("AzureAd", options));
            });

            services.AddRazorPages().AddMvcOptions(options =>
            {
                var policy = new AuthorizationPolicyBuilder()
                    .RequireAuthenticatedUser()
                    .Build();
                options.Filters.Add(new AuthorizeFilter(policy));
            });
```

3. In appsettings.json add this section, please use your Azure AAD details:

```JSON
"AzureAd": {
    "Instance": "https://login.microsoftonline.com/",
    "Domain": "<yourdomain>.onmicrosoft.com",
    "TenantId": "<yourTenantId>",
    "ClientId": "<yourClientId>",
    "CallbackPath": "/signin-oidc"
  },
```

4. Now try acceding to localhost:port/Module2/hello and you will see the error
![image](https://user-images.githubusercontent.com/46158478/127863943-30155814-76e5-4062-baf3-02010a1fa01b.png)
5. I've tried with removing these two lines, because there already existe in the orchard config:
```C#
            services.AddAuthentication(AzureADDefaults.AuthenticationScheme)
            .AddAzureAD(options => Configuration.Bind("AzureAd", options));
```
But, it show the folowing error:
![image](https://user-images.githubusercontent.com/46158478/127864616-14a04359-a97a-40e0-b03a-e7b05af03e89.png)

Could you help please

**Comment by Skrypt** [comment]
This is an issue with how you register these services. Could be the order or because you are not taking in consideration that we register these services on each tenants somewhere else.

**Comment by ns8482e** [comment]
`AddOrchardCore().AddMvc()` configures `MVC` and  `RazorPages` and `Authentication` for tenant pipeline, so you don't need to explicitly call ` services.AddMvc` , `services.AddAuthentication` and `services.AddRazorPages` on Application Services at host level instead you use Tenant level services. 

So it should be something like following.

```cs
            services.AddOrchardCore().AddMvc().ConfigureServices(svc =>
            {
                svc.AddAuthentication(options => options.DefaultSignOutScheme = AzureADDefaults.AuthenticationScheme)
                    .AddAzureAD(options => Configuration.Bind("AzureAd", options));
                svc.Configure<RazorPagesOptions>(options => {
                   var policy = new AuthorizationPolicyBuilder()
                    .RequireAuthenticatedUser()
                    .Build();
                    options.Filters.Add(new AuthorizeFilter(policy));
                });
   
            });
```

**Reply by omarhassani** [reply]
This solution crach because RazorPagesOptions deasn't have a definition for Filters.
I fixe this issue by removing the AddAuthentication and AddAzureAD registration from orchard :

                services.AddMvc(options => options.EnableEndpointRouting = false);
                services.AddAuthentication(AzureADDefaults.AuthenticationScheme)
                             .AddAzureAD(options => Configuration.Bind("AzureAd", options)); ;
                services.AddOrchardCore();
                services.AddRazorPages().AddMvcOptions(options =>
                {
                    var policy = new AuthorizationPolicyBuilder()
                        .RequireAuthenticatedUser()
                        .Build();
                    options.Filters.Add(new AuthorizeFilter(policy));
                });

Now it working, but I don't understand why adding AddAzureAD inside the AddOrchardCore confige cause the 401 error.

THank a lot for you suggestion
Best regards
Omar hassani
